go/vt/binlog/keyrange_filter.go

```diff
@@ -40,40 +30,23 @@ func KeyRangeFilterFunc(kit key.KeyspaceIdType, keyrange *pb.KeyRange, sendReply
 				log.Warningf("Not forwarding DDL: %s", statement.Sql)
 				continue
 			case proto.BL_DML:
-				keyspaceIndex := strings.LastIndex(statement.Sql, KEYSPACE_ID_COMMENT)
-				if keyspaceIndex == -1 {
+				keyspaceId, unfriendly, err := annotations.ParseSQLAnnotation(string(statement.Sql))
+				if err != nil {
 					updateStreamErrors.Add("KeyRangeStream", 1)
-					log.Errorf("Error parsing keyspace id: %s", statement.Sql)
+					log.Errorf(
+						"Error parsing keyspace id annotation. Skipping statement: %s, (%s)",
+						string(statement.Sql), err)
 					continue
 				}
-				idstart := keyspaceIndex + len(KEYSPACE_ID_COMMENT)
-				idend := strings.Index(statement.Sql[idstart:], SPACE)
-				if idend == -1 {
+				if unfriendly {
 					updateStreamErrors.Add("KeyRangeStream", 1)
-					log.Errorf("Error parsing keyspace id: %s", statement.Sql)
+					log.Errorf(
```

Comment:
Let's move that discussion to a cleanup bug, to figure out what we wanna do then. The behavior of this code as it is now is about the same as before the change, but feels clearer on what's happening, so it's a step forward. So let's submit this and move on. Probably the filtered replication part can be sent to @michael-berlin for later consideration.